src/twister2/plugin.py

```diff
@@ -42,7 +43,7 @@ def pytest_addoption(parser: pytest.Parser):
     twister_group = parser.getgroup('Twister')
     twister_group.addoption(
         '--build-only',
-        default=False,
+        default='--prep-artifacts-for-testing' in sys.argv,
```

Comment:
really strange dependency .. maybe better to move it to `validate_options` or `pytest_configure` and explain in comments, why `--buiild-only` is set

Reply:
I removed this `default='--prep-artifacts-for-testing' in sys.argv` completely from `--build-only` option. I believe that when users would like to achieve  this result, they can as well run Twister with arguments like:
`--build-only --prep-artifacts-for-testing`
and for that is much more clearer than enforcing silently `--build-only` when `--prep-artifacts-for-testing` is used (as it is done now in v1)